src/Migrate.php

```diff
@@ -120,6 +158,70 @@ private function restoreDestinationProject(array $restoreCredentials): void
         $this->logger->info('Current project restored');
     }
 
+    private function migrateSecrets(): void
+    {
+        $this->logger->info('Migrating secrets in configurations');
+
+        $sourceClient = $this->createSourceClient();
+
+        $sourceDevBranches = new DevBranches($sourceClient);
+        $sourceBranches = $sourceDevBranches->listBranches();
+        $defaultSourceBranch = current(array_filter($sourceBranches, fn($b) => $b['isDefault'] === true));
+
+        $sourceComponentsApi = new Components($sourceClient);
+        $components = $sourceComponentsApi->listComponents();
+        if (!$components) {
+            $this->logger->info('There are no components to migrate.');
+            return;
+        }
+
+        $encryptionApiUrl = $sourceClient->getServiceUrl('encryption');
+        $migrations = $this->createMigrationsClient($encryptionApiUrl);
+
+        foreach ($components as $component) {
+            if (in_array($component['id'], self::OBSOLETE_COMPONENTS, true)) {
+                continue;
+            }
+
+            foreach ($component['configurations'] as $config) {
+                $response = $migrations
+                    ->migrateConfiguration(
+                        $this->sourceProjectToken,
+                        Utils::getStackFromProjectUrl($this->destinationProjectUrl),
+                        $this->destinationProjectToken,
+                        $component['id'],
+                        $config['id'],
+                        (string) $defaultSourceBranch['id'],
+                    );
+
+                $this->logger->debug($response['message']);
+            }
+        }
+
+        $this->logger->info('Secrets in configurations have been migrated.');
+    }
+
+    private function createSourceClient(): StorageClient
+    {
+        if (isset($this->sourceClientFactory)) {
+            return (fn(): StorageClient => ($this->sourceClientFactory)())();
+        }
+        return new StorageClient(['token' => $this->sourceProjectToken, 'url' => $this->sourceProjectUrl ]);
+    }
+
+    private function createMigrationsClient(string $encryptionApiUrl): Migrations
+    {
```

Comment:
Tohleto mě teda přijde extrémně prasoidní, ono to znamená, že vlastně testy projdou s něčím úplně jiným než co se pak ve skutečnosti volá.
A taky mi to přijde celkem zbytečný - i StorageClient i Migrations by se klidně mohli předat v konstruktoru Migrate a svět by se nezbořil

Reply:
Ten zlý, ošklivý... nepěkná věc už je pryč. 🙂